components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtils.java

```diff
@@ -112,4 +113,20 @@ public static DCRMClientException generateClientException(DCRMConstants.ErrorMes
 
         return IdentityException.error(DCRMClientException.class, error.toString(), errorDescription);
     }
+
+    /**
+     * Validate application name according to the regex
+     *
+     * @return validated or not
+     */
+    public static boolean isRegexValidated(String applicationName) {
```

Comment:
You can reuse util methods from https://github.com/wso2/carbon-identity-framework/blob/5d4a622d69937b2032dc975fbe87eec4e543c7a8/components/identity-core/org.wso2.carbon.identity.base/src/main/java/org/wso2/carbon/identity/base/IdentityValidationUtil.java#L27-L27

instead of writing a new one.

Reply:
It is really complicated to use this, because it is validation reg-ex which are only available in the IdentityValidationUtil class. This make unit test to be re-write again which is an extra overhead. Therefore, IAM team decided not to use  IdentityValidationUtil class